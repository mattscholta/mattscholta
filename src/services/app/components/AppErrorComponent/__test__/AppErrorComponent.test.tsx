import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import {
  AppErrorComponent,
  AppErrorComponentProps
} from '../AppErrorComponent';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      asPath: '',
      pathname: '',
      query: {},
      route: '/'
    };
  }
}));

describe('AppErrorComponent', () => {
  let props: AppErrorComponentProps;
  let wrapper: RenderResult;

  const createWrapper = () => <AppErrorComponent {...props} />;

  beforeAll(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('returns null when there is no message', () => {
    expect(wrapper.container.firstChild).toBeNull();
    expect(wrapper.container.childNodes).toMatchSnapshot('null');
  });

  it('renders an error message', () => {
    props.message = 'Sample error message';
    wrapper.rerender(createWrapper());

    expect(wrapper.findAllByText(props.message));
    expect(wrapper.container.childNodes).toMatchSnapshot('rendered');
  });
});
