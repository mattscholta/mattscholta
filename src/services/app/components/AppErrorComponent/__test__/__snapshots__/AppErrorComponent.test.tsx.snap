// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppErrorComponent renders an error message: rendered 1`] = `
NodeList [
  <div
    class="u-center u-text-center u-block u-mx-auto u-my-4x u-text-14"
    style="max-width: 300px;"
  >
    <h3>
      Sorry, we've encountered an error.
    </h3>
    <p>
      Sample error message
    </p>
    <a
      href="/"
    >
      Go to the Homepage
    </a>
    <div
      class="u-flex u-flex-col u-gap-2x u-text-center u-m-2x"
    />
  </div>,
]
`;

exports[`AppErrorComponent returns null when there is no message: null 1`] = `NodeList []`;
